File: src/Exercise100.java

public class Exercise100 {
    public class TreeNode {
        int val;
        TreeNode left;
        TreeNode right;

        TreeNode() {
        }

        TreeNode(int val) {
            this.val = val;

        }

        TreeNode(int val, TreeNode left, TreeNode right) {
            this.val = val;
            this.left = left;
            this.right = right;
        }
    }

    class Solution {
        public boolean isSameTree(TreeNode p, TreeNode q) {
            if (p == null && q == null)
                return true;
            else if ((p != null && q != null) && p.val == q.val)
                return isSameTree(p.left, q.left) && isSameTree(p.right, q.right);
            else
                return false;

        }

    }
    
        public static void main(String[] args) {
        Exercise100 exercise = new Exercise100();
        Solution solution = exercise.new Solution();

        TreeNode node1_3 = exercise.new TreeNode(3);
        TreeNode node1_2 = exercise.new TreeNode(2, node1_3, null);
        TreeNode node1_1 = exercise.new TreeNode(1, null, node1_2);
       
        TreeNode node2_3 = exercise.new TreeNode(3);
        TreeNode node2_2 = exercise.new TreeNode(2, node2_3, null);
        TreeNode node2_1 = exercise.new TreeNode(1, null, node2_2);

        System.out.println(solution.isSameTree(node1_1, node2_1));
    }
}

File: src/BinaryTree/Exercise100.cpp
#include <iostream>
#include "BinaryTree.h"
using namespace std;

class Solution
{
public:
    bool isSameTree(TreeNode *p, TreeNode *q)
    {
        if (!p && !q)
            return true;
        else if (p && q)
            return p->val == q->val && isSameTree(p->left, q->left) && isSameTree(p->right, q->right);
        else
            return false;
    }
};

int main(){
    TreeNode a5(7);
    TreeNode a4(15);
    TreeNode a3(20, &a4, &a5);
    TreeNode a2(9);
    TreeNode a1(3, &a2, &a3);

    TreeNode b5(7);
    TreeNode b4(15);
    TreeNode b3(20, &b4, &b5);
    TreeNode b2(9);
    TreeNode b1(3, &b2, &b3);
    Solution sl;
    cout << sl.isSameTree(&a1,&b1);
}